app/client/src/widgets/SelectWidget/widget/propertyUtils.ts

```
@@ -218,6 +239,27 @@ export function valueKeyValidation(
 
     options = sourceData.map((d: Record<string, unknown>) => d[value]);
   } else if (_.isArray(value)) {
+    /*
+     * Here assumption is that if evaluated array is all equal, then it is a key,
+     * and we can return the parsed value(from source data) as the options.
+     */
+    const areAllValuesEqual = value.every((item, _, arr) => item === arr[0]);
+
+    if (
+      areAllValuesEqual &&
+      props.sourceData[0].hasOwnProperty(String(value[0]))
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)